### ISSUE TITLE
Validator builder do not pass constraint arguments to the constructor

### ISSUE THREAD

**Issue by adam-webee** [magento/magento2#31976]
### Preconditions (*)
1. Bug is reproducible on any Magento 2.4 or Magento 2.3.

When creating object instances with the object manager, parameters for the constructor have to be provided as an associative array, where the key represents the name of the parameter and the value represents its value. When 'validate.xml' file is parsed (in \Magento\Framework\Validator\Config) constraint arguments are parsed to an indexed array. The object manager does not support indexed arrays, so constraint arguments are not passed to the instantiated validator.
Validators that have mandatory constructor arguments failing to instantiate by throwing an exception about missing parameters.

Solution proposal in https://github.com/adam-webee/magento2/tree/bugfix/constraint-arguments-not-passed-by-object-manager.

### Steps to reproduce (*)
1. Create validation.xml file (e.g. WeBee/Bug/etc/validation.xml):
```xml
<validation xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:Validator/etc/validation.xsd">
    <entity name="bug_test">
        <rules>
            <rule name="rule1">
                <property_constraints>
                    <property name="name">
                        <constraint class="Magento\Framework\Validator\StringLength">
                            <argument>
                                <option name="min">5</option>
                                <option name="max">10</option>
                            </argument>
                        </constraint>
                    </property>
                </property_constraints>
            </rule>
        </rules>
        <groups>
            <group name="create">
                <uses>
                    <use rule="rule1"/>
                </uses>
            </group>
        </groups>
    </entity>
</validation>
```
2. Now invoke validator builder, e.g. this way:
```php
<?php

namespace WeBee\Bug\Model;

class CreateValidatorBug
{
    public function __construct(\Magento\Framework\Validator\Factory $factory)
    {
        $validator = $factory->createValidator('bug_test', 'create');
    }
}
```
3.  Debug Magento\Framework\Validator\StringLength constructor arguments - nothing has been provided.

### Expected result (*)
1. Magento\Framework\Validator\StringLength should receive such an array:
```php
['min' => 5, 'max' => 10]
```

### Actual result (*)
1. Magento\Framework\Validator\StringLength receives nothing

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [*] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._

**Comment by adam-webee** [comment]
@magento I am working on this

**Comment by adam-webee** [comment]
@magento add to contributors team

**Comment by adam-webee** [comment]
Hi,
This issue has not been yet confirmed and processed on the Magento side. I even proposed a solution - so please do not close it, as it is still present.

**Comment by engcom-Hotel** [comment]
Hi @adam-webee 

We have tried to reproduce this issue but stuck in 3rd step.We request you to please share a snapshot for the same.

**Comment by engcom-Hotel** [comment]
Hi @adam-webee ,

I hope that your issue is resolved because we did not get any update on this since more than 14 days. Hence closing the ticket.

Please feel free to raise the new ticket if you have any other issue. We will be happy to work for you.

**Comment by adam-webee** [comment]
Sorry for the late answer. I don't get it. Point number 3 in my issue description is: `Debug Magento\Framework\Validator\StringLength constructor arguments - nothing has been provided.`. You want me to include screens from debugging? 

To verify it - just run the code I provided in the description. Add a breakpoint inside the constructor for `Magento\Framework\Validator\StringLength` and you will see that there are missing arguments. 

Also, check the solution I provided to fix this issue.
